### DISCUSSION TITLE
dotnet-core-docker-and-cultures-solving-culture-issues-porting-a-net-core-app-from-windows-to-linux/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#318] [Announcements]
# dotnet-core-docker-and-cultures-solving-culture-issues-porting-a-net-core-app-from-windows-to-linux/

In this post I describe how I fixed an issue with Cultures that arose when porting an ASP.NET Windows app to ASP.NET Core on Linux.

https://andrewlock.net/dotnet-core-docker-and-cultures-solving-culture-issues-porting-a-net-core-app-from-windows-to-linux/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 13 2019, 10:39, in reply to [Havoc](https://disqus.com/by/jonasbark/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/318#discussioncomment-6058003)</em>

---

<p>I'm with you here. The big issue in this case was build images being different to production images which was very hard to debug.</p><p>Since then I ran into a _different_ cultures issue in another app, again due to the globalization invariant mode. Luckily my go-to assumption was this, so didn't waste too much time, but still... At this point, I will always make a point of installing the cultures in alpine for .NET. The bugs you can get are just too insidious and hard to pin down for the small size increase to be worth it IMO.</p><p>It seems like a fundamental difference between the bit of success and optimising for performance. Sure, those 30mb _may_ be something you need to squeeze out. But for the majority of users I suspect it's a timebomb waiting to catch out some developer later in the app cycle.</p><p>Producing images both with and without, would be good, but I'd rather they changed the defaults to include the cultures. If you're advanced enough to know about the trade offs, you're advanced enough to create a docker file without them too :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 17 2020, 08:19, in reply to [Lucas Vieira](https://disqus.com/by/disqus_Xf2gVHZW1H/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/318#discussioncomment-6058007)</em>

---

<p>Thanks! Glad to help 😀</p>

[a comment by an automated account is not shown]

**Comment by 2019Anton** [comment]
Thank you a lot! It solved my problem with aspnet:8.0-alpine and Only the invariant culture is supported in globalization-invariant mode. (Parameter 'name') en-us is an invalid culture identifier.
